library-client-rtsp/src/main/java/com/alexvas/rtsp/codec/FrameQueue.kt

```diff
@@ -21,19 +21,30 @@ class FrameQueue(frameQueueSize: Int) {
         if (queue.offer(frame, 5, TimeUnit.MILLISECONDS)) {
             return true
         }
+        if (queue.size >= frameQueueSize) {
+            Log.w(TAG, "Cannot add frame, queue is full. Queue Size :${queue.size}")
+            Thread.currentThread().interrupt()
+            queue.poll()
+            queue.clear()
+        }
         Log.w(TAG, "Cannot add frame, queue is full")
         return false
     }
 
     @Throws(InterruptedException::class)
     fun pop(): Frame? {
         try {
-            val frame: Frame? = queue.poll(1000, TimeUnit.MILLISECONDS)
-            if (frame == null) {
-                Log.w(TAG, "Cannot get frame, queue is empty")
+            if (queue.size>0) {
+                val frame: Frame? = queue.poll(1000, TimeUnit.MILLISECONDS)
+
+                if (frame == null) {
+                    Log.w(TAG, "Cannot get frame, queue is empty. Queue Size :${queue.size}")
+                }
+                return frame
             }
-            return frame
         } catch (e: InterruptedException) {
+            Log.i(TAG, "pop: InterruptedException caught. Queue Size: ${queue.size} ", e)
```

Comment:
Change:
1. Log.i -> Log.w
2. text to "InterruptedException caught. Queue size: ${queue.size}"